runner/app/routes/frame_interpolation.py

```diff
@@ -0,0 +1,122 @@
+    video: Annotated[UploadFile, File(description="Video file of any arbitrary length.")]= None,
+    inter_frames: Annotated[int, Form(description="Number of frames to create as the intermediate frames.")] = 2,
+    pipeline: Pipeline = Depends(get_pipeline),
+    token: HTTPAuthorizationCredentials = Depends(HTTPBearer(auto_error=False)),
+):
+    auth_token = os.environ.get("AUTH_TOKEN")
+    if auth_token:
+        if not token or token.credentials != auth_token:
+            return JSONResponse(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                headers={"WWW-Authenticate": "Bearer"},
+                content=http_error("Invalid bearer token"),
+            )
+        
+    if model_id != "" and model_id != pipeline.model_id:
+        return JSONResponse(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            content=http_error(
+                f"pipeline configured with {pipeline.model_id} but called with "
+                f"{model_id}"
+            ),
+        )
+    
+    # Prepare directories for input and output
+    temp_input_dir = "temp_input"
+    temp_output_dir = "temp_output"
+    os.makedirs(temp_input_dir, exist_ok=True)
+    os.makedirs(temp_output_dir, exist_ok=True)
+
+    try:
+        # Extract frames from video
+        video_data = await video.read()
+        frames = video_shredder(video_data, is_file_path=False)
+        # Save frames to temporary directory
+        for i, frame in enumerate(frames):
+            try:
+                frame_path = os.path.join(temp_input_dir, f"{i}.png")
+                cv2.imwrite(frame_path, frame)
+            except Exception as e:
+                logger.error(f"Error saving frame {i}: {e}")
+
+        # Create DirectoryReader and DirectoryWriter
+        reader = DirectoryReader(temp_input_dir)
+        writer = DirectoryWriter(temp_output_dir)
+
+        # Initialize FILMPipeline
+        pipeline(reader=reader, writer=writer, inter_frames=inter_frames)
+
+        writer.close()
+        reader.reset()
+
+        # Collect output frames
+        output_frames = []
+        path_to_file = sorted(
+            glob.glob(os.path.join(temp_output_dir, "*")),
+            key=lambda x: (int(os.path.basename(x).split(".")[0]), x)
+        )
+        for frame_in_path in path_to_file:
+            try:
+                frame = Image.open(frame_in_path)
+                output_frames.append(frame)
+            except Exception as e:
+                logger.error(f"Error reading frame {frame_in_path}: {e}")
+
+        # Wrap output frames in a list of batches (with a single batch in this case)
+        output_images = [[{"url": image_to_data_url(frame), "seed": 0, "nsfw": False} for frame in output_frames]]
+
+    except Exception as e:
+        logger.error(f"FILMPipeline error: {e}")
+        logger.exception(e)
+        return JSONResponse(
```

Comment:
Same comment here as on LivePortrait. :

I think it would be good if you could add the usage of handle_pipeline_exception here after rebasing or merging changes from main ( a follow up PR if you prefer  - example : https://github.com/livepeer/ai-worker/blob/726f7e22a3fafc5a3750c2e9ff089691fd5e4f87/runner/app/routes/image_to_text.py#L113 )